apps/rig_inbound_gateway/lib/rig_inbound_gateway/api_proxy/serializer.ex

```diff
@@ -33,18 +35,35 @@ defmodule RigInboundGateway.ApiProxy.Serializer do
   end
 
   # Search if header has key with given value
-  @spec header_value?(%Plug.Conn{}, String.t, String.t) :: boolean
-  def header_value?(conn, key, value) do
-    conn
-    |> Map.get(:resp_headers)
-    |> Enum.find({}, fn(headers_tuple) ->
+  @spec header_value?(headers_list, String.t, String.t) :: boolean
+  def header_value?(headers, key, value) do
+    headers
+    |> Enum.find({}, fn({header_key, _}) -> header_key == key end)
+    |> Tuple.to_list
+    |> Enum.member?(value)
+  end
+
+  # Transform keys for headers to lower-case
+  @spec downcase_headers(headers_list) :: headers_list
+  def downcase_headers(headers) do
+    headers
+    |> Enum.map(fn({key, value}) ->
       key_downcase =
-        headers_tuple
-        |> elem(0)
+        key
         |> String.downcase
-      key_downcase == key
+      {key_downcase, value}
     end)
-    |> Tuple.to_list
-    |> Enum.member?(value)
+  end
+
+  # Add new headers and update existing ones
+  @spec add_headers(headers_list, headers_list) :: headers_list
+  def add_headers(new_headers, old_headers) do
+    old_headers
+    |> Enum.filter(fn({old_key, _}) ->
+      new_headers
+      |> Enum.find(nil, fn({new_key, _}) -> old_key == new_key end)
+      |> is_nil
+    end)
+    |> Enum.concat(new_headers)
```

Comment:
hm maybe I get this wrong, but it seems this
1. takes all old headers that don't occur in `new_headers` and
2. add the new headers to that list.
Would that be easier to read if written this way:
```elixir
Map.merge(Map.new(old_headers), Map.new(new_headers))
|> Enum.to_list()
```

Reply:
Definitely, totally overlooked it.